packages/amplify_auth_cognito/lib/src/CognitoSession/AWSCognitoUserPoolTokens.dart

```diff
@@ -23,18 +23,20 @@ class AWSCognitoUserPoolTokens {
   String idToken;
   String refreshToken;
   AWSCognitoUserPoolTokens.init({@required LinkedHashMap<dynamic, dynamic> tokens}) {
-    var realTokens;
-    if (Platform.isAndroid) {
-      if (tokens.containsKey("value")) {
-        realTokens = tokens["value"];
+    if (tokens != null) {
+      var realTokens;
+      if (Platform.isAndroid) {
+        if (tokens.containsKey("value")) {
+          realTokens = tokens["value"];
+        } else {
+          throw(AmplifyDartExceptions.formatException(methodName: "fetchAuthSession", field: "credentials"));
+        }
       } else {
-        throw(AmplifyDartExceptions.formatException(methodName: "fetchAuthSession", field: "credentials"));
+        realTokens = tokens;
       }
```

Comment:
Shouldn't we be doing this in the android plugin code? If we intent to keep any logic here we should unit test it.

Reply:
Fixed this; moved the logic to platform code (Android)